src/dataProviders/csvProvider.ts

```diff
@@ -0,0 +1,82 @@
+import csv from 'csvtojson';
+import {Cluster, Itur, Coordinate, Rule, RuleValue, Constitution} from '../types';
+
+const readCsvFile = async (filePath: string): Promise<any[]> => {
+  const data: any[] = await csv().fromFile(filePath);
+
+  return data;
+};
+
+export const readClusters = async (): Promise<Cluster[]> => {
+  const clustersCsvData = await readCsvFile('./data/Tvirim.csv');
+
+  const clusters = clustersCsvData.map((clusterData) => createCluster(clusterData));
+
+  return clusters;
+};
+
+export const readIturim = async (): Promise<Itur[]> => {
+  const IturimCsvData = await readCsvFile('./data/iturim.csv');
+
+  const iturim: Itur[] = IturimCsvData.map((iturData) => createItur(iturData));
+
+  return iturim;
+};
+
+export const readConstitution = async (): Promise<Constitution> => {
+  const rulesCsvData = await readCsvFile('./data/Constitution.csv');
+  const constitution: Constitution = {};
```

Comment:
If we decide to use a record (object) instead of Map as a dictionary, we should create it with Object.create(null), in order to avoid the default keys colliding with our keys.
We might want to use Map in the future since it might perform better when adding the entries dynamically to the map.

Reply:
1. https://stackoverflow.com/questions/251402/create-an-empty-object-in-javascript-with-or-new-object
Here it says that their is no benefit of using `Object.create()` over `{}`, it's cleaner and readable, and their is no difference.
2. We can use `Map` in the future. For the simplicity of the program I prefer using dictionary for now, and if we will need better performance we will change it to `Map`. Moreover, since we load the rules "before the real program starts" it may not affect the performance on runtime (note the difference between `runtime` and `preprocess`)

Reply:
I agree there is no difference between {} and object.create(). But there is a difference between object.create() and object.create(null). object.create(null) does not have object prototype, so it is truelly empty without any properties or methods. Take a look here https://stackoverflow.com/questions/15518328/creating-js-object-with-object-createnull